renku/ui/api/models/project.py

```diff
@@ -69,6 +81,24 @@ def path(self):
         """Absolute path to project's root directory."""
         return self._client.path.resolve()
 
+    def status(
+        self, paths: Optional[List[Union[Path, str]]] = None, ignore_deleted: bool = False
+    ) -> Tuple[Dict[str, Set[str]], Dict[str, Set[str]], Set[str], Set[str]]:
+        """Return status of a project.
+
+        Args:
+            paths(Optional[List[Union[Path, str]]]): Limit the status to this list of paths (Default value = None).
+            ignore_deleted(bool): Whether to ignore deleted generations.
+
+        Returns:
+            Tuple[Dict[str, Set[str]], Dict[str, Set[str]], Set[str], Set[str]]: A quadruple containing a mapping of
+                stale outputs to modified inputs, a mapping of stale activities that have no generation to modified
+                inputs, a set of modified inputs, and a set of deleted inputs.
+
+        """
+        result = get_status_command().build().execute(paths=paths, ignore_deleted=ignore_deleted)
```

Comment:
This will create a new LocalClient based on CWD when we already have one in `self._client`. I don't think that can cause problems, but I wonder if we should add a way to set a client in the CommandBuilder directly? Or at least set the path?

That would also allow us to refactor https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/ui/service/controllers/api/mixins.py#L314-L315 to not use click context. Using click to inject a created client in the service is a bit ugly. But changing the service could be done in a follow-up issue.